src/modules/cpu/kernel/fog.hpp

```diff
@@ -0,0 +1,1315 @@
+                fogIntensityMaskPtrRow += fogWidth;
+            }
+        }
+
+        // Fog without fused output-layout toggle (NCHW -> NHWC)
+        else if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NCHW) && (dstDescPtr->layout == RpptLayout::NHWC))
+        {
+            Rpp8u *srcPtrRowR, *srcPtrRowG, *srcPtrRowB, *dstPtrRow;
+            Rpp32f *fogAlphaMaskPtrRow, *fogIntensityMaskPtrRow;
+            srcPtrRowR = srcPtrChannel;
+            srcPtrRowG = srcPtrRowR + srcDescPtr->strides.cStride;
+            srcPtrRowB = srcPtrRowG + srcDescPtr->strides.cStride;
+            dstPtrRow = dstPtrChannel;
+            fogAlphaMaskPtrRow = fogAlphaMaskPtr;
+            fogIntensityMaskPtrRow = fogIntensityMaskPtr;
+            for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+            {
+                Rpp8u *srcPtrTempR, *srcPtrTempG, *srcPtrTempB, *dstPtrTemp;
+                Rpp32f *fogAlphaMaskPtrTemp, *fogIntensityMaskPtrTemp;
+                srcPtrTempR = srcPtrRowR;
+                srcPtrTempG = srcPtrRowG;
+                srcPtrTempB = srcPtrRowB;
+                dstPtrTemp = dstPtrRow;
+                fogAlphaMaskPtrTemp = fogAlphaMaskPtrRow;
+                fogIntensityMaskPtrTemp = fogIntensityMaskPtrRow;
+
+                int vectorLoopCount = 0;
+#if __AVX2__
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrementPerChannel)
+                {
+                    __m256 pFogAlphaMask[2], pFogIntensityMask[2], p[6];
+                    rpp_simd_load(rpp_load16_f32_to_f32_avx, fogAlphaMaskPtrTemp, pFogAlphaMask);    // simd loads
+                    rpp_simd_load(rpp_load16_f32_to_f32_avx, fogIntensityMaskPtrTemp, pFogIntensityMask);    // simd loads
+                    rpp_simd_load(rpp_load48_u8pln3_to_f32pln3_avx, srcPtrTempR, srcPtrTempG, srcPtrTempB, p);    // simd loads
+                    compute_fog_48_host(p, pFogAlphaMask, pFogIntensityMask, pGamma);    // fog adjustment
+                    rpp_simd_store(rpp_store48_f32pln3_to_u8pkd3_avx, dstPtrTemp, p);    // simd stores
+                    srcPtrTempR += vectorIncrementPerChannel;
+                    srcPtrTempG += vectorIncrementPerChannel;
+                    srcPtrTempB += vectorIncrementPerChannel;
+                    dstPtrTemp += vectorIncrement;
+                    fogAlphaMaskPtrTemp += vectorIncrementPerChannel;
+                    fogIntensityMaskPtrTemp += vectorIncrementPerChannel;
+                }
+#endif
+                for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                {
+                    Rpp32f alphaFactor = ( 1 - (*fogAlphaMaskPtrTemp + gamma));
+                    Rpp32f intensityFactor = (*fogIntensityMaskPtrTemp * (*fogAlphaMaskPtrTemp + gamma));
+                    dstPtrTemp[0] = static_cast<Rpp8u>(RPPPIXELCHECK(std::nearbyintf(static_cast<Rpp32f>(*srcPtrTempR) * alphaFactor + intensityFactor)));
```

Comment:
Same comments on incrementing pointers directly here instead of doing explicit increment at end
Here we can increment srcPtrTempR, srcPtrTempG, srcPtrTempB like that
dstPtrTemp[0] = static_cast<Rpp8u>(RPPPIXELCHECK(std::nearbyintf(static_cast<Rpp32f>(*srcPtrTempR++) * alphaFactor + intensityFactor)));